diff --git a/win_build/boinc_vs2010.sln b/win_build/boinc_vs2010.sln
index 58cca98d8a..b3e4bc7b57 100644
--- a/win_build/boinc_vs2010.sln
+++ b/win_build/boinc_vs2010.sln
@@ -500,9 +500,7 @@ Global
 		{F243B93C-73CB-44E7-9BDC-847BB95C27CA}.EmuRelease|x64.ActiveCfg = Release|x64
 		{F243B93C-73CB-44E7-9BDC-847BB95C27CA}.EmuRelease|x64.Build.0 = Release|x64
 		{F243B93C-73CB-44E7-9BDC-847BB95C27CA}.Release|Win32.ActiveCfg = Release|Win32
-		{F243B93C-73CB-44E7-9BDC-847BB95C27CA}.Release|Win32.Build.0 = Release|Win32
 		{F243B93C-73CB-44E7-9BDC-847BB95C27CA}.Release|x64.ActiveCfg = Release|x64
-		{F243B93C-73CB-44E7-9BDC-847BB95C27CA}.Release|x64.Build.0 = Release|x64
 		{D9AF7F68-B881-45B1-A41C-B10E61D764EF}.Debug|Win32.ActiveCfg = Debug|Win32
 		{D9AF7F68-B881-45B1-A41C-B10E61D764EF}.Debug|x64.ActiveCfg = Debug|x64
 		{D9AF7F68-B881-45B1-A41C-B10E61D764EF}.EmuDebug|Win32.ActiveCfg = Release|Win32
@@ -606,9 +604,7 @@ Global
 		{25662612-421F-42F5-B5E1-D69ECBF3F5BB}.EmuRelease|x64.ActiveCfg = Release|x64
 		{25662612-421F-42F5-B5E1-D69ECBF3F5BB}.EmuRelease|x64.Build.0 = Release|x64
 		{25662612-421F-42F5-B5E1-D69ECBF3F5BB}.Release|Win32.ActiveCfg = Release|Win32
-		{25662612-421F-42F5-B5E1-D69ECBF3F5BB}.Release|Win32.Build.0 = Release|Win32
 		{25662612-421F-42F5-B5E1-D69ECBF3F5BB}.Release|x64.ActiveCfg = Release|x64
-		{25662612-421F-42F5-B5E1-D69ECBF3F5BB}.Release|x64.Build.0 = Release|x64
 		{49723CA5-DA05-43C0-93AB-6FD30D046919}.Debug|Win32.ActiveCfg = Debug|Win32
 		{49723CA5-DA05-43C0-93AB-6FD30D046919}.Debug|Win32.Build.0 = Debug|Win32
 		{49723CA5-DA05-43C0-93AB-6FD30D046919}.Debug|x64.ActiveCfg = Debug|x64
@@ -622,9 +618,7 @@ Global
 		{49723CA5-DA05-43C0-93AB-6FD30D046919}.EmuRelease|x64.ActiveCfg = Release|x64
 		{49723CA5-DA05-43C0-93AB-6FD30D046919}.EmuRelease|x64.Build.0 = Release|x64
 		{49723CA5-DA05-43C0-93AB-6FD30D046919}.Release|Win32.ActiveCfg = Release|Win32
-		{49723CA5-DA05-43C0-93AB-6FD30D046919}.Release|Win32.Build.0 = Release|Win32
 		{49723CA5-DA05-43C0-93AB-6FD30D046919}.Release|x64.ActiveCfg = Release|x64
-		{49723CA5-DA05-43C0-93AB-6FD30D046919}.Release|x64.Build.0 = Release|x64
 	EndGlobalSection
 	GlobalSection(SolutionProperties) = preSolution
 		HideSolutionNode = FALSE
