diff --git a/api/Makefile.am b/api/Makefile.am
index 6d13d3dc39..a51859d96d 100644
--- a/api/Makefile.am
+++ b/api/Makefile.am
@@ -42,20 +42,20 @@ AM_CXXFLAGS += @GLUT_CFLAGS@
 endif

 lib_LTLIBRARIES = libboinc_api.la
-pkgconfig_DATA = libboinc_api.pc
+## pkgconfig_DATA = libboinc_api.pc
 libboinc_api_la_SOURCES = $(api_files)
 libboinc_api_la_LDFLAGS = -version-number $(LIBBOINC_VERSION)

 if BUILD_GRAPHICS_API
 lib_LTLIBRARIES += libboinc_graphics2.la
-pkgconfig_DATA += libboinc_graphics2.pc
+## pkgconfig_DATA += libboinc_graphics2.pc
 libboinc_graphics2_la_SOURCES = $(graphics2_files)
 libboinc_graphics2_la_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/lib -I$(top_srcdir)/samples/image_libs
 libboinc_graphics2_la_LDFLAGS = -version-number $(LIBBOINC_VERSION) -ljpeg
 endif #BUILD_GRAPHICS_API

 lib_LTLIBRARIES += libboinc_opencl.la
-pkgconfig_DATA += libboinc_opencl.pc
+## pkgconfig_DATA += libboinc_opencl.pc
 libboinc_opencl_la_SOURCES = $(opencl_files)
 libboinc_opencl_la_LDFLAGS = -version-number $(LIBBOINC_VERSION)

diff --git a/configure.ac b/configure.ac
index 5ea74f247c..78db44401e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -64,8 +64,8 @@ SAH_LINKS
 AC_LANG_PUSH(C)
 AM_PROG_CC_C_O

-PKG_PROG_PKG_CONFIG
-PKG_INSTALLDIR
+dnl PKG_PROG_PKG_CONFIG
+dnl PKG_INSTALLDIR

 m4_divert_once([HELP_ENABLE],
   AS_HELP_STRING([BOINC Default enable values], [--enable-server --enable-client --enable-libraries --enable-manager: builds server, client, and libraries]))
diff --git a/lib/Makefile.am b/lib/Makefile.am
index 7cbcf7f000..9178459af4 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -180,7 +180,7 @@ noinst_HEADERS = \
     unix_util.h

 lib_LTLIBRARIES = libboinc.la
-pkgconfig_DATA = libboinc.pc
+## pkgconfig_DATA = libboinc.pc
 libboinc_la_SOURCES = $(generic_sources) $(mac_sources) $(win_sources)
 libboinc_la_CFLAGS = $(AM_CFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS)
 libboinc_la_CXXFLAGS = $(AM_CXXFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS)
@@ -192,7 +192,7 @@ libboinc_la_LIBADD =

 if ENABLE_BOINCCRYPT
 lib_LTLIBRARIES += libboinc_crypt.la
-pkgconfig_DATA += libboinc_crypt.pc
+## pkgconfig_DATA += libboinc_crypt.pc
 libboinc_crypt_la_SOURCES = crypt.cpp
 libboinc_crypt_la_CFLAGS = $(AM_CFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS) $(SSL_CFLAGS)
 libboinc_crypt_la_CXXFLAGS = $(AM_CXXFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS) $(SSL_CXXFLAGS)
@@ -202,7 +202,7 @@ endif

 if ENABLE_FCGI
 lib_LTLIBRARIES += libboinc_fcgi.la
-pkgconfig_DATA += libboinc_fcgi.pc
+## pkgconfig_DATA += libboinc_fcgi.pc
 libboinc_fcgi_la_SOURCES = $(libfcgi_sources) $(mac_sources) $(win_sources)
 libboinc_fcgi_la_CFLAGS = -D_USING_FCGI_ $(AM_CFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS)
 libboinc_fcgi_la_CXXFLAGS = -D_USING_FCGI_ $(AM_CXXFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS)
diff --git a/zip/Makefile.am b/zip/Makefile.am
index dde17a3fb5..5703274fff 100644
--- a/zip/Makefile.am
+++ b/zip/Makefile.am
@@ -60,7 +60,7 @@ libboinc_zip_sources += \
 endif

 lib_LTLIBRARIES = libboinc_zip.la
-pkgconfig_DATA = libboinc_zip.pc
+## pkgconfig_DATA = libboinc_zip.pc
 libboinc_zip_la_SOURCES = $(libboinc_zip_sources)
 libboinc_zip_la_LDFLAGS = -version-number $(LIBBOINC_VERSION)
 libboinc_zip_la_LIBADD =
