Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>// Top-level build file where you can add configuration options common to all sub-projects/modules.\r\nbuildscript {\r\n    ext.kotlin_version = '1.4.0'\r\n    repositories {\r\n        jcenter()\r\n        maven {\r\n            url 'https://maven.google.com/'\r\n            name 'Google'\r\n        }\r\n        google()\r\n    }\r\n    dependencies {\r\n        classpath 'com.android.tools.build:gradle:4.0.1'\r\n        classpath \"org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version\"\r\n        classpath 'com.vanniktech:gradle-android-junit-jacoco-plugin:0.16.0'\r\n    }\r\n}\r\n\r\nallprojects {\r\n    repositories {\r\n        jcenter()\r\n        maven {\r\n            url 'https://maven.google.com/'\r\n            name 'Google'\r\n        }\r\n    }\r\n}\r\n\r\napply plugin: \"com.vanniktech.android.junit.jacoco\"\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- build.gradle	(revision 9c90ee95e1c3f647d5cfe0b46b9b784cbe672a7d)
+++ build.gradle	(date 1602595194241)
@@ -10,7 +10,7 @@
         google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.0.1'
+        classpath 'com.android.tools.build:gradle:4.1.0'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
         classpath 'com.vanniktech:gradle-android-junit-jacoco-plugin:0.16.0'
     }
