diff --git a/images/apache/Dockerfile b/images/apache/Dockerfile
index b948c03..2354963 100644
--- a/images/apache/Dockerfile
+++ b/images/apache/Dockerfile
@@ -2,12 +2,12 @@
 ARG TAG
 
 #=====================================
-FROM php:7.0.31-apache-stretch AS base
+FROM php:7.3.33-apache-buster AS base
 #=====================================
 
 LABEL maintainer="Marius Millea <mariusmillea@gmail.com>"
 
-# install packages 
+# install packages
 RUN apt-get update \
     && apt-get install -y --no-install-recommends \
         cron \
@@ -15,7 +15,7 @@ RUN apt-get update \
         inotify-tools \
         libjpeg62-turbo-dev \
         libpng-dev \
-        libmariadbclient18 \
+        libmariadb3 \
         nano \
         openssl \
         python \
@@ -51,9 +51,9 @@ FROM base AS base-b2d
 # install Docker client
 RUN curl -L http://get.docker.com/builds/Linux/x86_64/docker-1.10.3 > /usr/bin/docker \
     && chmod +x /usr/bin/docker
-    
-    
-    
+
+
+
 #======================
 FROM base$TAG AS apache
 #======================
diff --git a/images/makeproject/Dockerfile b/images/makeproject/Dockerfile
index 75f9567..e71119b 100644
--- a/images/makeproject/Dockerfile
+++ b/images/makeproject/Dockerfile
@@ -2,7 +2,7 @@
 ARG TAG
 
 #===============================
-FROM debian:stretch-slim AS base
+FROM debian:buster-slim AS base
 #===============================
 
 LABEL maintainer="Marius Millea <mariusmillea@gmail.com>"
@@ -13,17 +13,17 @@ RUN apt-get update \
         ca-certificates \
         curl \
         dh-autoreconf \
-        g++ \ 
+        g++ \
         git \
         libcurl4-gnutls-dev \
         default-libmysqlclient-dev \
         libssl-dev \
         m4 \
         make \
-        mysql-client \
-        php7.0-cli \
-        php7.0-mysql \
-        php7.0-xml \
+        default-mysql-client \
+        php7.3-cli \
+        php7.3-mysql \
+        php7.3-xml \
         pkg-config \
         python \
         python3 \
@@ -31,7 +31,7 @@ RUN apt-get update \
         python3-mysqldb \
     && rm -rf /var/lib/apt/lists
 
-# get source and compile server 
+# get source and compile server
 COPY --chown=1000 boinc /usr/local/boinc
 RUN cd /usr/local/boinc && ./_autosetup && ./configure --disable-client --disable-manager && make
 
@@ -45,7 +45,7 @@ COPY --chown=1000 html /.project_root/html/
 COPY --chown=1000 secrets.env /run/secrets/
 
 #==============================
-FROM debian:stretch-slim AS b2d
+FROM debian:buster-slim AS b2d
 #==============================
 
 # do boinc2docker as a separate stage so we don't have to keep re-downloading
@@ -76,7 +76,7 @@ RUN apt-get update \
         wget \
     && rm -rf /var/lib/apt/lists
 
-COPY --from=b2d --chown=1000 /root/boinc2docker $HOME/boinc2docker 
+COPY --from=b2d --chown=1000 /root/boinc2docker $HOME/boinc2docker
 ENV PATH=$HOME/boinc2docker:$PATH
 
 
@@ -97,7 +97,7 @@ ONBUILD ENV BOINC_USER=$BOINC_USER \
             USER=$BOINC_USER \
             HOME=/home/$BOINC_USER \
             MYSQL_HOST=mysql
-    
+
 # set up the non-root user who compiles the server
 ONBUILD RUN adduser $BOINC_USER --disabled-password --gecos ""
 
